### ISSUE TITLE
metrics: set up alerting

### ISSUE THREAD

**Issue by cirocosta** [concourse/hush-house#26]
Hey,

Despite the fact that we already have metrics being collected by [Prometheus](https://prometheus.io/), and dashboards being displayed from Grafana, we're still in the need of having alerting set up in order to not need to constantly look at dashboards to know when things go wrong.

From my understanding, we have at least two choices here:

- go with [`grafana` alerts](https://grafana.com/docs/alerting/rules/) (like we do for [Wings](https://wings.pivotal.io)), or
- go with [`alertmanager`](https://prometheus.io/docs/alerting/alertmanager/), which would be notified by Prometheus when a particular query goes beyond a certain level.

While the second case seems interesting to me from the standpoint that we can decouple the visualization from the alerting, the second case allows us to visualize how our thresholds look like, as well as consume multiple data sources (e.g., not only Prometheus but [Stackdriver](https://cloud.google.com/stackdriver/docs/) as well).

**Acceptance criteria**:

- Can we use Prometheus for all alerting needs? 
- have the full alerting system working: having the observed metrics going up, we really get paged.

Thanks!

**Comment by scottietremendous** [comment]
We'll probably want alerts for Datadog SLIs/SLOs

**Comment by scottietremendous** [comment]
Chore: Bump Grafana to 6.0

**Comment by YoussB** [comment]
note: the domain doesn't redirect http requests to https

**Comment by taylorsilva** [comment]
It seems possible to use Prometheus with stackdriver, though the integration has some limitations and is in beta ([see here](https://cloud.google.com/monitoring/kubernetes-engine/prometheus)). Based on one [medium blog post](https://medium.com/google-cloud/prometheus-and-stackdriver-fb8f7524ece0), seems possible though!

Our current plan is to setup alerts with Grafana and create another long-term issue to implement alerts with Prometheus. We don't want to tackle prometheus now because we have a lot of learning to do about it. We'd rather get hush-house production-ready sooner rather than later, which means using Grafana for alerts in the short-term.

Thoughts?
cc @YoussB @cirocosta @scottietremendous

**Comment by cirocosta** [comment]
sounds goood 😁  thankss!

**Comment by scottietremendous** [comment]
@taylorsilva - I agree with this. We have other potential avenues to dive into Prometheus and hush-house doesn't need to be held back because of it.

**Comment by YoussB** [comment]
### EOD

- Added two alerts using Grafana. One for high disk usage and a second for high number of containers
- We setup PagerDuty integration with hush-house's Grafana and for the previous two alerts

**Next Steps:**

- [x] Think about what kind of alerts we want for hush-house and add them using Grafana.
- ~~Figure out why Grafana's HTTP is not redirecting to HTTPS and make it redirect.~~

**Comment by scottietremendous** [comment]
Let's make a new issue around diving into what suite of metrics we can provide on hush house outside of what we currently offer on Wings.

**Comment by cirocosta** [comment]
Hey,

@pivotal-bin-ju  and I were looking at Grafana performing the redirect, and it turns out that Grafana itself doesn't do that (differently from Concourse, which does redirect automatically), meaning that we'd need something else to perform the redirects (like `nginx`).

We'll leave this to another issue in order to not be blocked on this (something that the other metrics dashboard doesn't provide anyway).

Thanks!
